tests/pipelines/stable_cascade/test_stable_cascade_decoder.py

```diff
@@ -0,0 +1,192 @@
+    callback_cfg_params = ["image_embeddings", "text_encoder_hidden_states"]
+
+    @property
+    def text_embedder_hidden_size(self):
+        return 32
+
+    @property
+    def time_input_dim(self):
+        return 32
+
+    @property
+    def block_out_channels_0(self):
+        return self.time_input_dim
+
+    @property
+    def time_embed_dim(self):
+        return self.time_input_dim * 4
+
+    @property
+    def dummy_tokenizer(self):
+        tokenizer = CLIPTokenizer.from_pretrained("hf-internal-testing/tiny-random-clip")
+        return tokenizer
+
+    @property
+    def dummy_text_encoder(self):
+        torch.manual_seed(0)
+        config = CLIPTextConfig(
+            bos_token_id=0,
+            eos_token_id=2,
+            projection_dim=self.text_embedder_hidden_size,
+            hidden_size=self.text_embedder_hidden_size,
+            intermediate_size=37,
+            layer_norm_eps=1e-05,
+            num_attention_heads=4,
+            num_hidden_layers=5,
+            pad_token_id=1,
+            vocab_size=1000,
+        )
+        return CLIPTextModelWithProjection(config).eval()
+
+    @property
+    def dummy_vqgan(self):
+        torch.manual_seed(0)
+
+        model_kwargs = {
+            "bottleneck_blocks": 1,
+            "num_vq_embeddings": 2,
+        }
+        model = PaellaVQModel(**model_kwargs)
+        return model.eval()
+
+    @property
+    def dummy_decoder(self):
+        torch.manual_seed(0)
+        model_kwargs = {
+            "c_cond": 128,
+            "block_repeat": [[1, 1, 1, 1], [3, 3, 2, 2]],
+            "c_hidden": [16, 32, 64, 128],
+            "dropout": [0.1, 0.1, 0.1, 0.1],
+            "nhead": [-1, -1, 1, 2],
+            "level_config": ["CT", "CT", "CTA", "CTA"],
+            "blocks": [[1, 1, 1, 1], [1, 1, 1, 1]],
+            "switch_level": None,
+            "c_in": 4,
+            "c_out": 4,
+            "c_clip_text_pooled": 32,
+        }
+
+        model = StableCascadeUnet(**model_kwargs)
+        return model.eval()
+
+    def get_dummy_components(self):
+        decoder = self.dummy_decoder
+        text_encoder = self.dummy_text_encoder
+        tokenizer = self.dummy_tokenizer
+        vqgan = self.dummy_vqgan
+
+        scheduler = DDPMWuerstchenScheduler()
+
+        components = {
+            "decoder": decoder,
+            "vqgan": vqgan,
+            "text_encoder": text_encoder,
+            "tokenizer": tokenizer,
+            "scheduler": scheduler,
+            "latent_dim_scale": 4.0,
+        }
+
+        return components
+
+    def get_dummy_inputs(self, device, seed=0):
+        if str(device).startswith("mps"):
+            generator = torch.manual_seed(seed)
+        else:
+            generator = torch.Generator(device=device).manual_seed(seed)
+        inputs = {
+            "image_embeddings": torch.ones((1, 4, 4, 4), device=device),
+            "prompt": "horse",
+            "generator": generator,
+            "guidance_scale": 1.0,
```

Comment:
can we add a test for `guidance_scale > 1.0` because it seems to not work at the moment:
```
2024-02-13 14:41:10.149 [stderr   ]     |   File "/root/.cache/isolate/virtualenv/99fa2be920dca4756ebdf91b49cd139bde0469729194c6b8d7f74aa6b2a6a18a/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
2024-02-13 14:41:10.149 [stderr   ]     |     return func(*args, **kwargs)
2024-02-13 14:41:10.149 [stderr   ]     |            ^^^^^^^^^^^^^^^^^^^^^
2024-02-13 14:41:10.149 [stderr   ]     |   File "/root/.cache/isolate/virtualenv/99fa2be920dca4756ebdf91b49cd139bde0469729194c6b8d7f74aa6b2a6a18a/lib/python3.11/site-packages/diffusers/pipelines/stable_cascade/pipeline_stable_cascade.py", line 380, in __call__
2024-02-13 14:41:10.149 [stderr   ]     |     predicted_latents = self.decoder(
2024-02-13 14:41:10.149 [stderr   ]     |                         ^^^^^^^^^^^^^
2024-02-13 14:41:10.149 [stderr   ]     |   File "/root/.cache/isolate/virtualenv/99fa2be920dca4756ebdf91b49cd139bde0469729194c6b8d7f74aa6b2a6a18a/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1511, in _wrapped_call_impl
2024-02-13 14:41:10.149 [stderr   ]     |     return self._call_impl(*args, **kwargs)
2024-02-13 14:41:10.149 [stderr   ]     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-02-13 14:41:10.149 [stderr   ]     |   File "/root/.cache/isolate/virtualenv/99fa2be920dca4756ebdf91b49cd139bde0469729194c6b8d7f74aa6b2a6a18a/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1520, in _call_impl
2024-02-13 14:41:10.149 [stderr   ]     |     return forward_call(*args, **kwargs)
2024-02-13 14:41:10.149 [stderr   ]     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-02-13 14:41:10.149 [stderr   ]     |   File "/root/.cache/isolate/virtualenv/99fa2be920dca4756ebdf91b49cd139bde0469729194c6b8d7f74aa6b2a6a18a/lib/python3.11/site-packages/diffusers/pipelines/stable_cascade/modeling_stable_cascade_common.py", line 304, in forward
2024-02-13 14:41:10.149 [stderr   ]     |     x = x + self.effnet_mapper(
2024-02-13 14:41:10.149 [stderr   ]     |         ^^^^^^^^^^^^^^^^^^^^^^^
2024-02-13 14:41:10.149 [stderr   ]     | RuntimeError: The size of tensor a (2) must match the size of tensor b (4) at non-singleton dimension 0
2024-02-13 14:41:10.149 [stderr   ]     +------------------------------------
```